src/extension.ts

```diff
@@ -79,21 +79,33 @@ export async function activate(context: vscode.ExtensionContext) {
   const graph = makeGraph(graphConstructors);
   const thatMark = new ThatMark();
   const testCaseRecorder = new TestCaseRecorder(context);
+
   const cursorlessRecordTestCaseDisposable = vscode.commands.registerCommand(
     "cursorless.recordTestCase",
     async () => {
-      console.log("Recording test case for next command");
-      testCaseRecorder.start();
+      if (testCaseRecorder.active) {
+        vscode.window.showInformationMessage("Stopped recording test cases");
+        testCaseRecorder.stop();
+      } else {
+        if (await testCaseRecorder.start()) {
```

Comment:
If I've understood code correctly, isn't it an error if `start` returns false? Why not just let it throw an exception instead?

Reply:
That's one option. My thought was that that the user may have changed his mind. you don't really start before you have actually selected a folder

Reply:
Ok I didn't review too closely just seemed like there were lots of falses thrown around where an exception might make more sense eg user is in wrong directory. I think for case where user presses escape agreed we can just silently bail

Reply:
I can add some exceptions for in those cases.